src/Domain/CampaignRepository.php

```diff
@@ -289,6 +289,68 @@ public function setFundRepository(FundRepository $fundRepository): void
         $this->fundRepository = $fundRepository;
     }
 
+    /**
+     * @param Salesforce18Id<Campaign> $campaignId
+     */
+    public function totalAmountRaised(Salesforce18Id $campaignId): Money
+    {
+        $donationQuery = $this->getEntityManager()->createQuery(
+            <<<'DQL'
+            SELECT donation.currencyCode, COALESCE(SUM(donation.amount), 0) as sum FROM MatchBot\Domain\Donation donation
+            JOIN MatchBot\Domain\Campaign campaign
```

Comment:
This feels like it might be worth a small hit in value-object safety to avoid a DB join. If callers passed campaigns' numeric ID, which I think they can get just as easily from memory, we could avoid joining to Campaign. Not a blocker.

Reply:
Yep I think that's a worthwhile optimisation, will do adjust this.

And I don't think I'll do it yet but we could get the similar safety making a value object that wraps an int, could be generic the same way that Salesforce18Id is.